test/test_external.py

```diff
@@ -417,5 +418,20 @@ def test_respect_queue_when_load_from_config():
             assert interface("bob") == "foo"
 
 
+def test_raise_value_error_when_api_name_invalid():
+    with pytest.raises(InvalidApiName):
+        demo = gr.Blocks.load(name="spaces/gradio/hello_world")
+        demo("freddy", api_name="route does not exist")
+
+
+def test_use_api_name_in_call_method():
+    # Interface
+    demo = gr.Blocks.load(name="spaces/gradio/hello_world")
+    assert isinstance(demo("freddy", api_name="predict"), str)
+
+    app = gr.Blocks.load(name="spaces/gradio/queue-benchmark")
+    assert isinstance(app("freddy", api_name="text"), str)
```

Comment:
Same as above. Instead of just checking the type, can we assert the correct value for a more stringent test? 

Also, I think it would be good to test with a different Space that (A) does not have a time.sleep (B) and has multiple functions that do different things to ensure that the correct function is being called

Reply:
This space has multiple functions. I'll make a copy without the sleep!

Reply:
Yup great but it would also be good if the functions were doing different things to ensure that the correct function is being called